libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -287,11 +306,21 @@ private ModuleEntitlements computeEntitlements(Class<?> requestingClass) {
     private ModuleEntitlements getModuleScopeEntitlements(
         Class<?> callerClass,
         Map<String, List<Entitlement>> scopeEntitlements,
-        String moduleName
+        String moduleName,
+        String component,
+        String entitlementType,
+        String entitlementInfo
```

Comment:
Can this be lazy to avoid having to calculating it on every check?

Reply:
Done and done!